gobblin-metastore/src/main/java/org/apache/gobblin/metastore/FsStateStore.java

```diff
@@ -217,7 +217,7 @@ public void putAll(String storeName, String tableName, Collection<T> states) thr
   }
 
   protected void renamePath(Path tmpTablePath, Path tablePath) throws IOException {
-    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath);
+    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath, true);
```

Comment:
We are changing the behavior of renamePath(). I am assuming you need this to overwrite the watermarks.

Reply:
Yes, Fs state store does not accept add/update watermark into an existing store now, but in our case, we need to continually publish new file and update watermark so I change the behavior to make it able to overwrite existing state file.  Do you have any concern for this change? i.e we have pipeline that rely on this feature? If so, I can also make this behavior configurable